src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.cs

```diff
@@ -19,7 +20,7 @@ public static string FrameworkDescription
         {
             get
             {
-                return FrameworkName;
+                return string.Format("{0} {1}", FrameworkName, typeof(object).GetTypeInfo().Assembly.GetName().Version.ToString());
```

Comment:
> for future coding practices, how do you decide when to cache and when not to?

Based on expected / measured usage patterns.  If you expect this method to only be invoked rarely, then caching doesn't make sense: the string object(s) will be allocated once and the work to compute those strings done once, and then there's no need to hold onto them and take up space in the process after that.  If you instead expect this method will be used repeatedly, then it makes more sense to cache rather than doing the work and allocating each time.  And if you expect it to be used really frequently, on a hot path, with every operation mattering, you might even choose to initialize the static as part of its declaration rather than lazily in the property so as to avoid branching costs to check for whether it's been initialized.

For this specific case, it definitely won't fall into the last case.  And I doubt it even falls into the middle case.  More than likely, caching here is unnecessary.  I was simply commenting that if something made sense to cache, it would be the result of the whole operation rather than just the version used in the operation.